partials/single-featured.php

```diff
@@ -0,0 +1,18 @@
+<?php
```

Comment:
I think the name `single-featured` is likely to be confused with single template for a (non existent) featured post type.

maybe `featured-post.php`